integration-tests/kubernetes/kubernetes.go

```diff
@@ -40,3 +40,18 @@ func CheckForPod(pods v1.PodList, targetRegex string) bool {
 
 	return false
 }
+
+// CheckPVC return true if number of pvcs in the list is equal to expected PVCs
+func CheckPVC(pvcs v1.PersistentVolumeClaimList, expectedPVC int) bool {
+	return len(pvcs.Items) == expectedPVC
+}
+
+// CheckPVCForBound return true if all the pvcs in the list are bound
+func CheckPVCForBound(pvcs v1.PersistentVolumeClaimList) bool {
```

Comment:
Idiomatic maya based solution where the core has most of the business logic:

```go
// pkg/kubernetes/persistentvolumeclaim/v1alpha1/pvc.go
type pvc struct {
  object *apis.PersistentVolumeClaim
}

type buildOption func(*pvc)

func WithObject(obj *apis.PersistentVolumeClaim) buildOption {
  return func(p *pvc) {
    p.object = obj
  }
}

func New(opts ...buildOption) *pvc {
  p := &pvc{}
  for _, o := range opts {
    o(p)
  }
  return p
}

type predicate func(*pvc) bool

func IsBound() predicate {
  return func(p *pvc) bool {
    return p.Status.Phase == "Bound"
  }
}

type listBuilder struct {
  list []*pvc
}

func ListBuilder() *listBuilder {
  return &listBuilder{}
}

func (b *listBuilder) WithListObject(list apis.PersistentVolumeClaimList) *listBuilder{
  // ...
}

func (l *pvcList) IsBound() bool {
  for _, pvc := l.list {
    if !IsBound(pvc) {
      return false
    }
  }
  return true
}
```

Reply:
cool added a new package for pvc